site/en/blog/extension-instantnav/index.md

```diff
@@ -0,0 +1,179 @@
+
+Chrome has been working hard at making navigation fast. Instant Navigation
+technologies such as [Back/Forward Cache](https://web.dev/bfcache/)
+([shipped](https://chromestatus.com/feature/6279906713403392) on desktop in M96)
+and [Speculation Rules](https://chromestatus.com/feature/5740655424831488)
+(shipped in M103) improve both the going back and going forward experience.
+In this post we will explore the updates we’ve made to browser extensions APIs
+to accommodate these new workflows.
+
+## Understanding the types of pages
+
+Prior to the introduction of Back/Forward Cache and Prerendering, an individual
+tab only had one active page. This was always the one that was visible. If a
+user returns to the previous page, the active page would be destroyed (Page B)
+and the previous page in history would be completely reconstructed (Page A).
+Extensions did not need to worry about what life cycle pages were in because
+there was only one for a tab, the Active/Visible state.
+
+{% Img src="image/zKrSUSkPboWMTTSEkowJbqw5Egi2/VR5hJLKud7lNIou8baHx.png", alt="Eviction of active page", width="350", height="209" %}
+
+With Back Forward Cache and Prerendering, there is no longer a 1:1 relationship
+between tabs and pages. Now, each tab actually stores multiple pages and pages
+transition between states rather than being destroyed and reconstructed.
+
+For example, a page could begin its life as a prerendered (not visible) page,
+transition into an active (visible) page when the user clicks a link, and then
+be stored in the Back Forward Cache (not visible) when the user navigates to
+another page, all without the page ever being destroyed. Later in this article
+we will look at the new properties exposed to help extensions understand what
+state pages are in.
+
+{% Img src="image/zKrSUSkPboWMTTSEkowJbqw5Egi2/0DT7Tzx0vQhNZgegsoEj.png", alt="Types of Pages", width="531", height="209" %}
+
+Note that a tab can have a series of prerender pages (not just one), a single
+*active* (visible) page, and a series of Back/Forward cached pages.
+
+## What’s changing for extension developers?
+### FrameId == 0
+In Chromium, we refer to the topmost/main frame as the outermost frame.
+
+Extension authors that assume the [*frameId*](/docs/extensions/reference/webNavigation/#a-note-about-frame-ids:~:text=onErrorOccurred%20event%20fired.-,frameId,-number)
+of the outermost [frame is 0](/docs/extensions/reference/webNavigation/#a-note-about-frame-ids)
+may have problems. Since a tab can now have multiple outermost frames
+(prerendered and cached pages), the assumption that there is a single outermost
+frame for a Tab is incorrect. frameId == 0 will still continue to represent the
+outermost frame of the **active** page, but the outermost frames of **other**
+pages will be non-zero. A new field frameType has been added to fix this
+problem. See the [“How do I determine if a frame is the outermost frame?”](#outermost-frame)
+section.
```

Comment:
@dtapuska, as you likely know based on the use of your text fragment link, there isn't a great way to link directly to the *frameId* property on the callback of `webNavigation.getAllFrames()`. There also isn't a great description of what this property is or how it works today. I think it may be best if we expand the reference documentation to explicitly describe the roles of *frameId* and *documentId*. 

```suggestion
Extension authors that assume the
[*frameId*]((/docs/extensions/reference/webNavigation/#:~:text=onErrorOccurred%20event%20fired.-,frameId,-number))
of the outermost frame is 0 (a previous best practice) may have problems. Since
a tab can now have multiple outermost frames (prerendered and cached pages), the
assumption that there is a single outermost frame for a Tab is incorrect.
`frameId == 0` will still continue to represent the outermost frame of the
**active** page, but the outermost frames of **other** pages in the same tab
will be non-zero. A new field *frameType* has been added to fix this problem.
See the [How do I determine if a frame is the outermost
frame?](#outermost-frame) section of this post.
```

I want to try to steer away from linking readers to the [A note about frame IDs](/docs/extensions/reference/webNavigation/#a-note-about-frame-id) section. It's not a great look to point to the live version of our own docs to highlight of date guidance. If we _do_ want to direct readers to out of date guidance, I'd be more comfortable linking to a Wayback Machine copy rather than the live site. 

I'd very much like to update the webNavigation API's guidance either in this PR or in another that follows shortly after. @dtapuska and @jpmedley, do you have a preference on which path we take? Given how long Dave has been waiting on this PR, I'm inclined to tackle it separately but I'm open to either.

Reply:
I'll take care of the API guidance. Let's not wait for it.